server/tinylicious/src/routes/ordering/documents.ts

```diff
@@ -72,5 +81,31 @@ export function create(storage: IDocumentStorage): Router {
 		);
 	});
 
+	/**
+	 * Passes on content to all clients in a collaboration session happening on the document via means of signal.
+	 */
+	router.post("/:tenantId/:id/broadcast-signal", (request, response) => {
+		const tenantId = getParam(request.params, "tenantId");
+		const documentId = getParam(request.params, "id");
+		const signalContent = getParam(request.body, "signalContent");
+		const documentP = storage.getDocument(tenantId, documentId);
+
+		documentP.then(
+			(_document: any) => {
+				try {
+					const signalRoom: IRoom = { tenantId, documentId };
+					const payload: IBroadcastSignalEventPayload = { signalRoom, signalContent };
+					collaborationSessionEventEmitter.emit("broadcast-signal", payload);
```

Comment:
`collaborationSessionEventEmitter` can be undefined. Use `collaborationSessionEventEmitter?.emit`